### ISSUE TITLE
comment_references should ignore escaped opening `[`

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#2442]
`\[` is an escaped opening bracket in markdown.

The linter should not consider such a bracket for the purpose of finding comment references.

**Comment by pq** [comment]
Can you give an example where we're flagging this wrong?

**Comment by kevmoo** [comment]
```dart
/// Examples: counter { metric: "/debug_access_count" field: "iam_principal" }
/// ==> increment counter /iam/policy/debug_access_count {iam_principal=\[value
/// of IAMContext.principal\]}
```

This is auto-generated for the googleapis package

I want to be able to put in (escaped) `[` characters

**Comment by pq** [comment]
Thanks!

**Comment by srawlins** [comment]
We might want to fix this with a big hammer, because I think there are many avoidable false positives in comment_references.

For a long time I've wanted to add an AST structure in the markdown package using SourceSpans. With this tooling, you could take a dartdoc comment like:

```markdown
/// [ref] and \[non-ref\]
/// and `code with [not a ref]`.
```
and get back an AST like `[ReferenceLink(label="ref"), TextNode(" \[non-ref\]\nand "), Code("code with [not a ref]")]`

You can then check all ReferenceLinks. When one is an unknown reference, you can look at `ReferenceLink.span.line`, etc. and report there.

**Comment by kevmoo** [comment]
I would LOVE that @srawlins

**Comment by natebosch** [comment]
It would also be great to support markdown named link reference. These are currently flagged but dartdoc links them correctly.
https://github.com/dart-lang/test/blob/d11240d2fb1f5abf8666d0c7acaa261ba937e051/pkgs/test_api/lib/src/backend/configuration/on_platform.dart#L9-L11
